service/src/main/java/io/camunda/service/license/CamundaLicense.java

```diff
@@ -37,12 +38,18 @@ public synchronized String getLicenseType() {
   }
 
   public synchronized void initializeWithLicense(final String license) {
-    if (!isInitialized) {
+    if (isInitialized) {
+      return;
+    }
+
+    if (license != null && !license.isBlank()) {
       isValid = determineLicenseValidity(license);
       licenseType = getLicenseTypeFromProperty(license);
-
-      isInitialized = true;
+    } else {
+      LOGGER.error("No license detected when one is expected. Please provide a license.");
```

Comment:
❓ If the license is not provided we logged an error. But what happens next ? It's not clear to me

Reply:
I added an action the customer is expected to take to fix the error in the message

But as to what will happen, the license validity will be marked as false, and the licenseType will be set to `unknown`. I added explicit setters to this block to make that clear